sycl/include/sycl/ext/oneapi/memory_export.hpp

```diff
@@ -0,0 +1,125 @@
+//==------------- memory_export.hpp --- SYCL memory export -----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <cstddef> // for size_t
+
+#include "interop_common.hpp"     // for external_mem_handle_type
+#include <sycl/context.hpp>       // for context
+#include <sycl/detail/export.hpp> // for __SYCL_EXPORT
+#include <sycl/device.hpp>        // for device
+#include <sycl/queue.hpp>         // for queue
+
+namespace sycl {
+inline namespace _V1 {
+namespace ext::oneapi::experimental {
+
+template <external_mem_handle_type ExternalMemHandleType> struct exported_mem;
+
+template <> struct exported_mem<external_mem_handle_type::opaque_fd> {
+  using type = int;
+};
+
+template <> struct exported_mem<external_mem_handle_type::win32_nt_handle> {
+  using type = void *;
+};
+
+template <external_mem_handle_type ExternalMemHandleType>
+using exported_mem_t = typename exported_mem<ExternalMemHandleType>::type;
```

Comment:
This is no longer part of the API, so move it to the `detail` namespace.

Reply:
Moved to `detail`.